cmd/registry/cmd/upload/bulk/protos.go

```diff
@@ -289,10 +305,115 @@ func (task *uploadProtoTask) fileName() string {
 
 func (task *uploadProtoTask) zipContents() ([]byte, error) {
 	prefix := task.directory + "/"
-	contents, err := core.ZipArchiveOfPath(task.path, prefix, true)
+
+	// Get the proto files in the main directory.
+	protos, err := localProtos(task.path, prefix)
+	if err != nil {
+		return nil, err
+	}
+
+	// Compile the listed protos to get their dependencies.
+	count := len(protos)
+	for count > 0 {
+		protos, err = referencedProtos(protos, task.directory)
+		if err != nil {
+			return nil, err
+		}
+		if len(protos) == count {
+			break
+		}
+		count = len(protos)
+	}
+
+	// Get the metadata files in the main directory.
+	metadata, err := localMetadata(task.path, prefix)
+	if err != nil {
+		return nil, err
+	}
+
+	allfiles := append(metadata, protos...)
```

Comment:
I think this would be fine to inline, but otherwise it should probably be `files` or `allFiles`.

Reply:
Inlined, thanks!